### PR TITLE
guard us from caller that give us missing data such as error id or fi…

### PR DESCRIPTION
**Pull request by heejaechang**
…le path.

fix https://github.com/dotnet/roslyn/issues/16612

### CODE REVIEW

**Comment by heejaechang** [comment]
@mavasani @srivatsn @davkean @panopticoncentral can you take a look? added simple null check.

**Comment by davkean** [comment]
So this is us passing though a null error id?

**Comment by heejaechang** [comment]
@davkean except the callstack, I dont know exactly who and why it gives null error id or file path to us. if you want to, we can add NFW to find out why/who.

except that, those 2 variables (error id and filepath seems only one that can cause null exception in the code and now I guarded them)

**Comment by heejaechang** [comment]
ping? can I get some code review?

File: src/VisualStudio/Core/Def/Implementation/TaskList/ProjectExternalErrorReporter.cs [hunk 1]
```diff
@@ -42,6 +42,13 @@ public ProjectExternalErrorReporter(ProjectId projectId, string errorCodePrefix,
 
         private bool CanHandle(string errorId)
         {
+            // make sure we have error id, otherwise, we simple don't support
+            // this error
+            if (errorId == null)
+            {
+                return false;
```
Comment (mavasani) [inline src/VisualStudio/Core/Def/Implementation/TaskList/ProjectExternalErrorReporter.cs]: Please log an NFW so we can fix the error source generating these corrupt diagnostics.

**Comment by heejaechang** [comment]
vsi is expected to fail.

**Pull request merged**